source/extensions/filters/http/composite/action.cc

```diff
@@ -9,6 +9,42 @@ void ExecuteFilterAction::createFilters(Http::FilterChainFactoryCallbacks& callb
   cb_(callbacks);
 }
 
+float ExecuteFilterActionFactory::getSampleRatio(
+    const envoy::extensions::filters::http::composite::v3::ExecuteFilterAction& composite_action) {
+  // In default case, if sample_percent is not populated, sample_ratio is 100%, i.e, all sampled.
+  if (!composite_action.has_sample_percent()) {
+    float all_sampled = 1;
+    return all_sampled;
+  }
+  const auto& sample_percent = composite_action.sample_percent();
+  float denominator = 100;
+  switch (sample_percent.denominator()) {
+  case envoy::type::v3::FractionalPercent::HUNDRED:
+    denominator = 100;
+    break;
+  case envoy::type::v3::FractionalPercent::TEN_THOUSAND:
+    denominator = 10000;
+    break;
+  case envoy::type::v3::FractionalPercent::MILLION:
+    denominator = 1000000;
+    break;
+  default:
+    throw EnvoyException(
+        fmt::format("ExecuteFilterAction sample_percent config denominator setting "
+                    "is invalid : {}. Valid range 0~2.",
+                    sample_percent.denominator()));
```

Comment:
PANIC_ON_PROTO_ENUM_SENTINEL_VALUES is defined as below:  https://github.com/envoyproxy/envoy/blob/de6b730c3f59e8ee3ee756c60bc826d7ede295e1/source/common/common/assert.h#L309

It can not capture a bad configuration, i.e, if denominator is set to a number , like 5.

Reply:
Catching invalid configuration like 5 should be PGV's responsibility, as `[(validate.rules).enum = {defined_only: true}];` is defined there.

Looks like PGV rule is not actually implemented here. I think it is safer to have this then before PGV is implemented